### PR TITLE
WIP: build using @theforeman/builder

### PR DESCRIPTION
**Pull request by sharvit**
Notice the build results are still not runnable.

Works with: https://github.com/theforeman/foreman-js/pull/71

### CODE REVIEW

**Comment by evgeni** [comment]
@sharvit should this be closed now that we merged https://github.com/theforeman/foreman/pull/7178? or is this the next step

**Comment by sharvit** [comment]
No, it’s the next step :)
Still experimental, but eventually I would like to have a module that responsible for building the client (js, css)

File: config/application.rb [hunk 1]
```diff
@@ -77,6 +77,20 @@ class Application < Rails::Application
...
       config.paths['config/routes.rb'] << route_file
     end
+    
+    config.webpack = ActiveSupport::OrderedOptions.new
+    config.webpack.dev_server = ActiveSupport::OrderedOptions.new
+    
+    config.webpack.dev_server.host = proc { respond_to?(:request) ? request.host : 'localhost' }
+    config.webpack.dev_server.port = 3808
+    
+    config.webpack.dev_server.https = false
```
Comment (xprazak2) [inline config/application.rb:87]: I remember there used to be problems related to http/https when running in vagrant box. I currently run webpack with:

```
./node_modules/.bin/webpack-dev-server --config config/webpack.config.js --https --key /etc/pki/katello/private/foreman-devel.example.com-foreman-client.key --cert /etc/pki/katello/certs/foreman-devel.example.com-foreman-client.crt --cacert /etc/pki/katello/certs/katello-default-ca.crt --watch --progress --public foreman-devel.example.com --port 3808
```

and have
```yml
:webpack_dev_server: true
:webpack_dev_server_https: true
```
in config/settings.yaml

File: lib/foreman/builder/helper.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+require 'action_view'
+require 'foreman/builder/manifest'
+
+module Foreman
+  module Builder
+    module Helper
+      def builded_js_for source
```
Comment (ekohl) [inline lib/foreman/builder/helper.rb:7]:
```suggestion
      def built_js_for source
```

**Pull request closed without merging**